pico/tools/src/main/java/io/helidon/pico/tools/TypeTools.java

```diff
@@ -1167,7 +1042,7 @@ static String extractInjectionPointTypeInfo(TypeSignature sig,
                 ClassRefTypeSignature typeArgSig = toClassRefSignature(csig.getTypeArguments().get(0), enclosingElem);
                 varTypeName = typeArgSig.toString();
                 handled = typeArgSig.getTypeArguments().isEmpty();
-                if (!handled && isProviderType(typeArgSig.getClassInfo().getName())) {
+                if (!handled && isProviderType(TypeName.create(typeArgSig.getClassInfo().getName()))) {
```

Comment:
and here

Reply:
Annotation processor does not need to be as performant as a server. Also `TypeName` is the type we should use to do comparison and checking of types, so this is the correct approach (there may be space for even more refactoring to use it more widely, so we can get rid of all string based comparisons and validations)

Reply:
I have done some refactoring in this area. Still using `TypeName`, just more efficiently in some cases.